src/track/beats.cpp

```diff
@@ -467,11 +604,31 @@ Beats::ConstIterator Beats::iteratorFrom(audio::FramePos position) const {
     } else {
         it = std::lower_bound(cfirstmarker(), clastmarker() + 1, position);
     }
-    DEBUG_ASSERT(it == cbegin() || it == cend() || *it >= position);
+    // Due to precision error, it happens that `it` and `position` are equal,
+    // but `it` is slightly behind `position`. To mitigate that, we aim for a
+    // 10e-3 precision
+    DEBUG_ASSERT(it == cbegin() || it == cend() || *it > position ||
+            std::fabs(*it - position) < 0.001);
```

Comment:
This is an existing problem on `main` - I had this assert going on randomly and couldn't come up as to what was wrong. I believe this was the reason

Reply:
Is there a reason to use the hardcoded constant of 0.001 instead of https://github.com/mixxxdj/mixxx/blob/e06279cedd921ec063a053e048b9dbdfed342a73/src/track/beats.cpp#L22

Reply:
Yes - me not being aware of it :) 
I will use that constant.